lib/app/connector.rb

```diff
@@ -0,0 +1,145 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License;
+# you may not use this file except in compliance with the Elastic License.
+#
+
+# frozen_string_literal: true
+
+require 'active_support'
+require 'concurrent'
+require 'connectors'
+require 'cron_parser'
+require 'utility'
+
+module App
+  module Connector
+    SYNC_JOB_POOL = Concurrent::ThreadPoolExecutor.new(
+      :min_threads => 8,
+      :max_threads => 8,
+      :max_queue => 1_000,
+      :fallback_policy => :abort
+    )
+    CONNECTORS_INDEX = '.elastic-connectors'
+    QUERY_SIZE = 20
+
+    @running = Concurrent::AtomicBoolean.new(false)
+
+    class << self
+
+      def start!
+        running!
+
+        Utility::Logger.info('Starting to process jobs.')
+        start_polling_jobs
+      end
+
+      def running?
+        running.true?
+      end
+
+      private
+
+      attr_reader :running
+
+      def running!
+        raise 'The connector app is already running!' unless running.make_true
+      end
+
+      def start_polling_jobs
+        loop do
+          polling_jobs
+        rescue StandardError => e
+          Utility::Logger.error("Error happened during sync. error: #{e.message}")
+          raise
```

Comment:
what happens in that case, do we die?

Reply:
No, it's supposed to just log the error and go to the next loop. I added this for testing purposes, but forgot to remove it 😓